ql/src/java/org/apache/hadoop/hive/ql/cache/results/QueryResultsCache.java

```diff
@@ -555,20 +558,20 @@ public boolean setEntryValid(CacheEntry cacheEntry, FetchWork fetchWork) {
           return false;
         }
 
-        if (requiresMove) {
-          // Move the query results to the query cache directory.
-          cachedResultsPath = moveResultsToCacheDirectory(queryResultsPath);
-          dataDirMoved = true;
+        if (requiresCaching) {
+          cacheEntry.cachedResultPaths = new HashSet<>();
+            for(FileStatus fs:fetchWork.getFilesToFetch()) {
+              cacheEntry.cachedResultPaths.add(fs);
+            }
         }
-        LOG.info("Moved query results from {} to {} (size {}) for query '{}'",
```

Comment:
It might be useful to keep the logging statement which gives the size of the results cache entry that was created here.

Reply:
Good point. Will updated the code.